mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/CompletedJobStore.java

```diff
@@ -0,0 +1,374 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.jobcluster;
+
+import com.netflix.spectator.api.BasicTag;
+import io.mantisrx.common.Label;
+import io.mantisrx.common.metrics.Metrics;
+import io.mantisrx.common.metrics.MetricsRegistry;
+import io.mantisrx.common.metrics.spectator.GaugeCallback;
+import io.mantisrx.common.metrics.spectator.MetricGroupId;
+import io.mantisrx.master.jobcluster.JobClusterActor.LabelCache;
+import io.mantisrx.master.jobcluster.job.IMantisJobMetadata;
+import io.mantisrx.master.jobcluster.job.JobState;
+import io.mantisrx.server.master.domain.JobClusterDefinitionImpl.CompletedJob;
+import io.mantisrx.server.master.domain.JobId;
+import io.mantisrx.server.master.persistence.MantisJobStore;
+import io.mantisrx.shaded.com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import lombok.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+interface ICompletedJobsStore {
+
+    /**
+     * Initialize the store
+     * @throws IOException if there is an error
+     */
+    void initialize() throws IOException;
+
+
+    /**
+     * Gets a completed job given a job id
+     * @param jId job id
+     * @return completed job if found else empty
+     * @throws IOException if there is an error
+     */
+    Optional<CompletedJob> getCompletedJob(JobId jId) throws IOException;
+
+    /**
+     * Gets a completed job given a job id
+     * @param jId job id
+     * @return completed job if found else empty
+     * @throws IOException if there is an error
+     */
+    Optional<IMantisJobMetadata> getJobMetadata(JobId jId) throws IOException;
+
+    /**
+     * Gets a list of completed jobs
+     * @param limit number of jobs to return
+     * @return list of completed jobs
+     * @throws IOException if there is an error
+     */
+    List<CompletedJob> getCompletedJobs(int limit) throws IOException;
+
+    /**
+     * Gets a list of completed jobs
+     * @param limit number of jobs to return
+     * @param endExclusive end job id
+     * @return list of completed jobs
+     * @throws IOException if there is an error
+     */
+    List<CompletedJob> getCompletedJobs(int limit, JobId endExclusive) throws IOException;
```

Comment:
So this only supports reading first x jobs? Is there a way to read from the list paginated?